src/main/java/com/amazon/ion/bytecode/ir/Operation.kt

```diff
@@ -0,0 +1,121 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package com.amazon.ion.bytecode.ir
+
+/**
+ * Defines operation codes for all Ion bytecode instructions.
+ *
+ * Operation codes are constructed by combining instruction kinds with variant identifiers.
+ * Each operation represents a specific instruction that can be executed by the Ion bytecode
+ * interpreter. Operations are organized by data type and functionality.
+ *
+ * See `com/amazon/ion/bytecode/ir/instruction_reference.md` for more details about the instruction set.
+ */
+internal object Operation {
+
+    /**
+     * Extracts the instruction kind from an operation code.
+     *
+     * @param operation The operation code to extract the instruction kind from
+     * @return The instruction kind portion of the operation code
+     */
+    @JvmStatic
+    fun toInstructionKind(operation: Int): Int = operation ushr OPERATION_KIND_OFFSET
```

Comment:
I don't care what name we choose, but the fact that `InstructionKind` is accessed by `OPERATION_KIND_OFFSET` shows that we need to pick one. Slight preference for `OperationKind`.